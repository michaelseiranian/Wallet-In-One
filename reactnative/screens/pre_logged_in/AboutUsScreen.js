import {
    StyleSheet,
    Text,
    TouchableOpacity,
    View,
    Image,
    ScrollView,
    ImageBackground,
  } from "react-native";
import React from "react";
import { useTheme } from 'reactnative/src/theme/ThemeProvider';
import {styles} from 'reactnative/screens/All_Styles.style.js';
  
export default function AboutUsScreen ({ navigation }) {

    const {dark, colors, setScheme} = useTheme();

    const stylesInternal = StyleSheet.create({
      aboutContainer: {
        display: "flex",
        flex: 1
      },
      background: {
          width: '100%',
          height: '100%'
      },
      logo:{
          width: 290,
          height: 280,
          marginTop: '-10%',
          alignSelf:'center'
      },
      imgStyle: {
        width: 370,
        height: 150,
        borderRadius: 40,
        marginTop: '-10%',
        alignSelf:'center'
      },
      paraStyle: {
        fontSize: 16,
        paddingBottom: 30,
      },
      aboutLayout: {
        backgroundColor: colors.primary,
        paddingHorizontal: 30,
        marginVertical: 30,
        paddingBottom: 10
      },
      aboutSubHeader: {
        fontSize: 18,
        color: "#fff",
        textTransform: "uppercase",
        fontWeight: "500",
        marginVertical: 15,
        alignSelf: "center",
      },
      aboutPara: {
        color: "#fff",
      },
      homepage: {
        backgroundColor: 'white',
        color: colors.primary,
        width: "75%",
        borderRadius: 25,
        textAlign: 'center',
        fontWeight: 'bold',
        marginLeft: '11%',
        padding: "2%",
        fontSize:  27,
        marginTop: '-2%'
      },
      developers: {
        backgroundColor: 'black',
        color: colors.primary,
        width: "40%",
        borderRadius: 25,
        textAlign: 'center',
        fontWeight: 'bold',
        padding: "2%",
        fontSize:  17,
        marginTop: '-4%',
        alignSelf: 'center',
      },
    });

    return (
      <ImageBackground
        source={require('reactnative/assets/background.jpg')}
        style={stylesInternal.background}
      >
      <ScrollView
      contentContainerStyle={{
        flexGrow : 1,
        justifyContent : 'center',
        paddingBottom: 20
        }}
      style={stylesInternal.aboutContainer}
      >
        <Image
            source={require('reactnative/assets/logo.png')}
            style={stylesInternal.logo}
            resizeMode="contain"
          >
        </Image>
        <View>
          <Image
            style={stylesInternal.imgStyle}
            source={require('reactnative/assets/wallets.webp')}
          />
        </View>
  
        <View
        contentContainerStyle={{
            paddingBottom: 30,
            }}
        style={stylesInternal.aboutLayout}
        
        >
          <Text style={stylesInternal.aboutSubHeader}> About us </Text>
          <Text style={[stylesInternal.paraStyle, stylesInternal.aboutPara]}>
            Welcome to Wallet-In-One! We are an all in one central finance
            app where you can access your credit cards, debit cards, stocks and
            crytocurrency. Simply connect your respective accounts and view
            the latest changes in your finances. {"\n"}
            This app was developed by second year students from King's College 
            London on the Computer Science course for the module 'Software
            Engineering Group Project' for a client. All IP belongs to the client
            and further development is to be continued by said client.
          </Text>
          <TouchableOpacity
            onPress={() => navigation.navigate('Developer Info')}
          >
             <Text style={stylesInternal.developers}>Meet the team!</Text>
          </TouchableOpacity>
        </View>
        
        <TouchableOpacity
            onPress={() => navigation.navigate('Start')}
        >
           <Text style={stylesInternal.homepage}>Home Page</Text>
        </TouchableOpacity>
  
      </ScrollView>
      </ImageBackground>
    );
  };