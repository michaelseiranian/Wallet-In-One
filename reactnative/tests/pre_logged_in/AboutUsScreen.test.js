import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import AboutUsScreen from '../../screens/pre_logged_in/AboutUsScreen';
import renderer from 'react-test-renderer';

describe('<AboutUsScreen/>', () => {
    it('developer team button test', () => {
        const navigate = jest.fn();
        const { getByText } = render(<AboutUsScreen  navigation={{ navigate }}/>);
        fireEvent.press(getByText('Meet the team!'));
        expect(navigate).toHaveBeenCalledWith('Developer Info');
    })

    it('start button test', () => {
        const navigate = jest.fn();
        const { getByText } = render(<AboutUsScreen  navigation={{ navigate }}/>);
        fireEvent.press(getByText('Home Page'));
        expect(navigate).toHaveBeenCalledWith('Start');
    })

    it('snapshot aboutUsScreen test', () => {
        const navigate = jest.fn();
        const aboutUsScreen = renderer.create(<AboutUsScreen navigation={{ navigate }} />);
        expect(aboutUsScreen).toMatchSnapshot();         
      });
})