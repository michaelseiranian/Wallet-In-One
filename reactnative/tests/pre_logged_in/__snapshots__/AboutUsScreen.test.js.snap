// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutUsScreen/> snapshot aboutUsScreen test 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "justifyContent": "center",
        "paddingBottom": 20,
      }
    }
    style={
      {
        "display": "flex",
        "flex": 1,
      }
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          {
            "alignSelf": "center",
            "height": 280,
            "marginTop": "-10%",
            "width": 290,
          }
        }
      />
      <View>
        <Image
          source={1}
          style={
            {
              "alignSelf": "center",
              "borderRadius": 40,
              "height": 150,
              "marginTop": "-10%",
              "width": 370,
            }
          }
        />
      </View>
      <View
        contentContainerStyle={
          {
            "paddingBottom": 30,
          }
        }
        style={
          {
            "backgroundColor": "purple",
            "marginVertical": 30,
            "paddingBottom": 10,
            "paddingHorizontal": 30,
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
              "marginVertical": 15,
              "textTransform": "uppercase",
            }
          }
        >
           About us 
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
                "paddingBottom": 30,
              },
              {
                "color": "#fff",
              },
            ]
          }
        >
          Welcome to Wallet-In-One! We are an all in one central finance app where you can access your credit cards, debit cards, stocks and crytocurrency. Simply connect your respective accounts and view the latest changes in your finances. 
          

          This app was developed by second year students from King's College London on the Computer Science course for the module 'Software Engineering Group Project' for a client. All IP belongs to the client and further development is to be continued by said client.
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "black",
                "borderRadius": 25,
                "color": "purple",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginTop": "-4%",
                "padding": "2%",
                "textAlign": "center",
                "width": "40%",
              }
            }
          >
            Meet the team!
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 25,
              "color": "purple",
              "fontSize": 27,
              "fontWeight": "bold",
              "marginLeft": "11%",
              "marginTop": "-2%",
              "padding": "2%",
              "textAlign": "center",
              "width": "75%",
            }
          }
        >
          Home Page
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
